portal-impl/src/com/liferay/portlet/internal/ActionRequestImpl.java

```diff
@@ -28,12 +42,66 @@
 
 	@Override
 	public ActionParameters getActionParameters() {
-		throw new UnsupportedOperationException();
+		return _actionParameters;
 	}
 
 	@Override
 	public String getLifecycle() {
 		return PortletRequest.ACTION_PHASE;
 	}
 
+	@Override
+	public void init(
+		HttpServletRequest request, Portlet portlet,
+		InvokerPortlet invokerPortlet, PortletContext portletContext,
+		WindowState windowState, PortletMode portletMode,
+		PortletPreferences preferences, long plid) {
+
+		super.init(
+			request, portlet, invokerPortlet, portletContext, windowState,
+			portletMode, preferences, plid);
+
+		Map<String, String[]> actionParameterMap = new LinkedHashMap<>();
+		Map<String, String[]> parameterMap = getParameterMap();
+		String portletNamespace = PortalUtil.getPortletNamespace(
+			getPortletName());
+		Map<String, String[]> servletRequestParameterMap =
+			request.getParameterMap();
+		RenderParameters renderParameters = getRenderParameters();
```

Comment:
@ngriffin7a this would cause UnsupportedOperationException, right? No one on the hierarchy is implementing getRenderParameters() method.

Reply:
@shuyangzhou Good catch. That's probably why there were so many failures from CI. I was just running "retest" to see if it was really my fault. I will send an updated PR that contains an implementation of `PortletRequestImpl.getRenderParameters()` that returns a non-null value for now. I am trying to keep these PRs as small as possible so as not to overwhelm you and Brian.